### PR TITLE
Add migraphx.create_argument to python API

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Review by shivadbhavsar — COMMENTED** [review]
Its already possible create an argument object with user-defined data in the current API using `migraphx.argument(...)`. Whats the use case for having this option, is there a case where np arrays aren't sufficient for doing this?

**Comment by pfultz2** [comment]
> Its already possible create an argument object with user-defined data in the current API using migraphx.argument(...). Whats the use case for having this option, is there a case where np arrays aren't sufficient for doing this?

This is to enable the driver output to create the same scalars when using `--py`(currently it just generates random values), and this output should work with only `import migraphx`, which is why using numpy wouldn't work.

**Review by shivadbhavsar — APPROVED** [review]

File: doc/src/reference/py.rst [hunk 1]
```diff
@@ -121,6 +121,15 @@ argument
 
     :rtype: argument
 
+.. py:function:: create_argument(s, values)
+
+    Create an argument of shape s with a set of values.
+
+    :param shape s: Shape of argument to create.
+    :param list values: Values to put in the argument. Must be the same number of elements as the shape.
```
Comment (umangyadav) [inline doc/src/reference/py.rst:129]: > Must be the same number of elements as the shape

This is not enforced. Assert inside `fill()` checks for `<=`.

**Review by CharlieL7 — APPROVED** [review]
Approved, should probably update the doc wrt Umang's comment.

File: doc/src/reference/py.rst (hunk 1 above)
Comment (pfultz2) [reply doc/src/reference/py.rst:129]: Added a check for the size.

File: src/py/migraphx_py.cpp [hunk 2]
```diff
@@ -547,6 +547,13 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
         py::arg("format") = "msgpack");
 
     m.def("get_target", &migraphx::make_target);
+    m.def("create_argument", [](const migraphx::shape& s, const std::vector<float>& values) {
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply src/py/migraphx_py.cpp]: Switched to use double.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**